### PR TITLE
Reducer tests

### PR DESCRIPTION
**Pull request by phil303**
Pretty self explanatory
:eyeglasses: @schwers

### CODE REVIEW

File: src/app/reducers/commentsPages.test.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+          params: { id: POST_ID },
+          postId: POST_ID,
+          loading: true,
+          loadingMoreChildren: {},
+          results: [],
+        };
+
+        // inject the cached comments page into the store ahead of time
+        const { store } = getStore();
+        store.getState().commentsPages = { '1': CACHED_COMMENTS_PAGE };
```
Comment (nramadas) [inline src/app/reducers/commentsPages.test.js]: couldn't you use the state initialization of getStore here?

Comment (phil303) [inline src/app/reducers/commentsPages.test.js]: This is a fairly invasive test and i'm not completely sold we should keep it . It's testing to make sure the reducer checks for cached data and doesn't do any unnecessary updates to the store. Therefore it's checking for identity - mocha's `equal` instead of value - mocha's `eql`.

Comment (phil303) [inline src/app/reducers/commentsPages.test.js]: And now that I think about it I can do it that way. I'll update it as such but I'm hoping someone convinces me this is a dumb test too

File: src/app/reducers/commentsPages.test.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        const LOADING_COMMENTS_PAGE = {
+          id: '1',
+          params: { id: POST_ID },
+          postId: POST_ID,
+          loading: true,
+          loadingMoreChildren: {},
+          results: [],
+        };
+        const RESULTS = [
+          { type: 'comment', uuid: 't1_comment1', paginationId: 't1_comment1' },
```
Comment (ajacksified) [inline src/app/reducers/commentsPages.test.js]: 🔕 it may be good to use a real `Record` object for these, so that as the API evolves we don't break all of our tests. It's also a slightly more concise syntax:

```
const RESULTS = [
  new Record('comment', 't1_comment1', 't1_comment1'),
  ...
];
```

We could be _even more_ pedantic and use the `TYPES` from https://github.com/reddit/node-api-client/blob/2ca06bf4ebf7bc7a15cd08c07cea897e1f3c1dff/apis/base.es6.js#L13 instead of hardcoding `t1`, but that's probably a bit over the top for something that's probably never going to change, unless you wanted to implement a `stubRecord(type, id=fakeid, paginationId=null)` method to build fake stubs in api-client.

File: src/app/reducers/accountRequests.test.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        expect(accountRequests).to.eql({});
+      });
+    });
+
+    describe('FETCHING_ACCOUNT', () => {
+      it('should add an account optimistically', () => {
+        const ACCOUNT = { id: 'foobar', loading: true, result: {} };
+        const { store } = getStore();
+        store.dispatch({
```
Comment (ajacksified) [inline src/app/reducers/accountRequests.test.js]: [prefer newlines around multiline blocks](https://github.com/reddit/styleguide/tree/master/javascript#18.7) when not the first block of its indentation level

File: src/app/reducers/commentsPages.test.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
+import createTest from '@r/platform/createTest';
+import merge from '@r/platform/merge';
+
+import commentsPages from './commentsPages';
+import * as commentsPageActions from 'app/actions/commentsPage';
+import * as loginActions from 'app/actions/login';
+import * as apiResponseActions from 'app/actions/apiResponse';
+
+
```
Comment (ajacksified) [inline src/app/reducers/commentsPages.test.js]: extra newline 😱

File: src/app/reducers/accountRequests.test.js (hunk 3 above)
Comment (phil303) [inline src/app/reducers/accountRequests.test.js]: oh interesting. Will update!

File: src/app/reducers/commentsPages.test.js (hunk 2 above)
Comment (schwers) [inline src/app/reducers/commentsPages.test.js]: FWIW `paginationId` is defaulted to `uuid` (src)[https://github.com/reddit/node-api-client/blob/3X/apiBase/Record.es6.js] so if you were using `Record`s it would be even more concise. e.g.
```javascript
const RESULTS = [
  new Record('comment', 't1_comment'),
  ...
];
```

File: src/app/reducers/commentsPages.test.js (hunk 4 above)
Comment (phil303) [inline src/app/reducers/commentsPages.test.js]: oh yea there's a few of these. That's a personal habit of mine to separate the imports from the business logic. I can remove.

File: src/app/reducers/collapsedComments.test.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import createTest from '@r/platform/createTest';
+
+import collapsedComments from './collapsedComments';
+import * as commentActions from 'app/actions/comment';
+import * as loginActions from 'app/actions/login';
+
+createTest({ reducers: { collapsedComments } }, ({ getStore, expect }) => {
+  describe('collapsedComments', () => {
+    describe('LOGGED_IN and LOGGED_OUT', () => {
+      it('should return default on log in', () => {
```
Comment (ajacksified) [inline src/app/reducers/collapsedComments.test.js]: What is the purpose of these loggedin/out tests? The collapse reducer isn't affected by login state.

File: src/app/reducers/commentsPages.test.js (hunk 2 above)
Comment (phil303) [inline src/app/reducers/commentsPages.test.js]: Yea. Using `Record` is the right approach - a lot of this was an exercise in understanding our codebase and there's probably a few spots like this. I'll update.

File: src/app/reducers/commentsPages.js [hunk 6]
```diff
@@ -29,16 +29,18 @@ export default (state=DEFAULT, action={}) => {
...
-    case commentsPageActions.RECEIEVED_COMMENTS_PAGE: {
+    case commentsPageActions.RECEIVED_COMMENTS_PAGE: {
       const { commentsPageId, commentsPageResults } = action;
       const currentCommentsPage = state[commentsPageId];
       if (!currentCommentsPage) { return state; }
 
       return merge(state, {
         [commentsPageId]: {
           loading: false,
+          // TODO: what happens if a user adds a comment before results come back?
```
Comment (ajacksified) [inline src/app/reducers/commentsPages.js]: Good point - we should probably use the `concat` strategy of `merge` here: https://github.com/reddit/node-platform/blob/master/lib/modules/merge.js

File: src/app/reducers/commentsPages.js [hunk 7]
```diff
@@ -29,16 +29,18 @@ export default (state=DEFAULT, action={}) => {
...
           results: commentsPageResults,
         },
+        // TODO: this feels inherently race condition-ish
```
Comment (ajacksified) [inline src/app/reducers/commentsPages.js]: agreed, this is a hack (I wrote it) - it would be better to send the current comment page ID with the newModel action instead, so it can update its page.

File: src/app/reducers/collapsedComments.test.js (hunk 5 above)
Comment (phil303) [inline src/app/reducers/collapsedComments.test.js]: uh, you're right. i can't figure out where the heck I got this.

File: src/app/reducers/commentsPages.test.js [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+        expect(commentsPages).to.eql({});
+      });
+    });
+
+    describe('FETCHING_COMMENTS_PAGE', () => {
+      it('should try the cache for a pre-existing commentsPage first', () => {
+        const POST_ID = 't3_12345';
+        const CACHED_COMMENTS_PAGE = {
+          id: '1',
```
Comment (ajacksified) [inline src/app/reducers/commentsPages.test.js]: it would be best to put the `'1'` into a const and reference it everywhere below (for this and all tests where data such as ids are re-used), so that tests are more refactorable

File: src/app/reducers/compact.test.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
+import createTest from '@r/platform/createTest';
+
+import * as compactAction from 'app/actions/compact';
+import * as loginActions from 'app/actions/login';
+import compact from './compact';
+
+createTest({ reducers: { compact } }, ({ getStore, expect }) => {
+  describe('compact', () => {
+    describe('LOGGED_IN and LOGGED_OUT', () => {
```
Comment (ajacksified) [inline src/app/reducers/compact.test.js]: (like my other comment, why logged in/out?)

File: src/app/reducers/collapsedComments.test.js (hunk 5 above)
Comment (schwers) [inline src/app/reducers/collapsedComments.test.js]: Technically it does (https://github.com/reddit/reddit-mobile/commit/f142b4593de4a7de9511f7f57bfd826704db87db). When I did auth stuff I added the log in / out clearing to all of the reducers that were there. In this case I could see the argument for preserving what's collapsed when you log-in. But clearing on log-out seems like the right idea to me personally, as I think it would be weird to log-out and see any remnants of what I did logged in other than links being a visited color (as that's standard browser behavior).

Comment (ajacksified) [inline src/app/reducers/collapsedComments.test.js]: Oh, you're right - I didn't notice that when looking through the code. Disregard!

File: src/app/actions/accounts.js [hunk 10]
```diff
@@ -7,8 +7,8 @@ const { AccountsEndpoint } = endpoints;
 export const FETCHING_ACCOUNT = 'FETCHING_ACCOUNT';
 export const fetching = (options) => ({ type: FETCHING_ACCOUNT, ...options });
 
-export const RECEIEVED_ACCOUNT = 'RECEIEVED_ACCOUNT';
```
Comment (schwers) [inline src/app/actions/accounts.js]: 🔕 Well, this is embarrassing. My editor tends to aggressively autocomplete across project files so I must have spelled received incorrectly early on and and just blissfully autocompleted it in multiple places. Thanks for fixing this!

Comment (ajacksified) [inline src/app/actions/accounts.js]: never had that problem with vim :trollface:

File: src/app/reducers/collapsedComments.test.js (hunk 5 above)
Comment (phil303) [inline src/app/reducers/collapsedComments.test.js]: ah thank god. thought i was losing it

File: src/app/reducers/commentsPages.test.js [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+          loadingMoreChildren: {},
+          results: [],
+        };
+
+        const { store } = getStore({
+          commentsPages: { '1': CACHED_COMMENTS_PAGE },
+        });
+
+        store.dispatch({
+          type: commentsPageActions.FETCHING_COMMENTS_PAGE,
```
Comment (schwers) [inline src/app/reducers/commentsPages.test.js]: Testing philosophy question: is there are reason to build the actions manually rather than use the action creators? e.g.
```javascript
store.dispatch(commentPageActions.fetching('1', { id: POST_ID }));
``` 
It seems like if we used the creators and then had separates tests to verify they're building the correct action objects the tests would be more concise and refactorable.

Comment (ajacksified) [inline src/app/reducers/commentsPages.test.js]: :+1:

Comment (phil303) [inline src/app/reducers/commentsPages.test.js]: Yea that's a good point. It'd exercise more code as well and that's only a good thing. TBH, I just started writing the tests and didn't slow down to think about that.

File: src/app/reducers/commentsPages.js (hunk 6 above)
Comment (phil303) [inline src/app/reducers/commentsPages.js]: @ajacksified Not a huge fan of leaving TODOs in code but I want to address this eventually and not in this patch. I can remove and add it to my own todo list or I can leave it. Do you have a preference here?

**Comment by phil303** [comment]
:haircut:

File: src/app/reducers/commentsPages.js (hunk 6 above)
Comment (ajacksified) [inline src/app/reducers/commentsPages.js]: Yeah, a fix should be a separate PR.

**Comment by ajacksified** [comment]
looks good to me. 🐟

thanks for splitting each reducer test into a separate commit! makes reviewing very nice.
